src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PractitionerMapper.java

```diff
@@ -1,32 +1,51 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
 import org.hl7.fhir.dstu3.model.Practitioner;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+
+import lombok.RequiredArgsConstructor;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
-import uk.nhs.digital.nhsconnect.lab.results.model.edifact.RequesterNameAndAddress;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.InvolvedParty;
-
-import java.util.Optional;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
 
 @Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 public class PractitionerMapper {
+    private final UUIDGenerator uuidGenerator;
+
     private static final String SDS_USER_SYSTEM = "https://fhir.nhs.uk/Id/sds-user-id";
 
     public Optional<Practitioner> mapRequester(final Message message) {
         return message.getInvolvedParties().stream()
             .map(InvolvedParty::getRequesterNameAndAddress)
             .flatMap(Optional::stream)
-            .map(this::toPractitioner)
+            .map(r -> mapToPractitioner(r.getIdentifier(), r.getRequesterName()))
             .findAny();
     }
 
-    private Practitioner toPractitioner(final RequesterNameAndAddress requester) {
+    public Optional<Practitioner> mapPerformer(final Message message) {
+        return message.getInvolvedParties().stream()
+            .map(InvolvedParty::getPerformerNameAndAddress)
+            .flatMap(Optional::stream)
+            .filter(p -> !StringUtils.isBlank(p.getIdentifier()))
```

Comment:
Can you add a test covering the scenario?

Reply:
Created this test in 9d42e9f